File: go/binary_tree_tilt.go
package leetcode

import (
	"math"
)

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func findTilt(root *TreeNode) int {
	_, res := dfs(root)
	return res
}

func dfs(node *TreeNode) (int, int) {
	if node == nil {
		return 0, 0
	}

	ls, lt := dfs(node.Left)
	rs, rt := dfs(node.Right)

	return ls + rs + node.Val, lt + rt + int(math.Abs(float64(ls-rs)))
}

File: python/binary-tree-tilt.py
from typing import Optional

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
        
class Solution:
    def findTilt(self, root: Optional[TreeNode]) -> int:
        def dfs(node):
            if not node:
                return (0, 0)
            
            ls, lt = dfs(node.left)
            rs, rt = dfs(node.right)
            
            return (ls+rs+node.val, lt+rt+(abs(ls-rs)))
        
        return dfs(root)[1]